packages/web-service/src/pages/eligibility/eligibility.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * Route handlers for the eligibility check flow
+ */
+import { yesNoPage, isYes } from '../common/yes-no.js'
+import { checkAnswersPage } from '../common/check-answers.js'
+import { eligibilityURIs } from '../../uris.js'
+import pageRoute from '../../routes/page-route.js'
+// The pages in the flow
+const {
+  LANDOWNER, LANDOWNER_PERMISSION, CONSENT, CONSENT_GRANTED,
+  NOT_ELIGIBLE_LANDOWNER, NOT_ELIGIBLE_PROJECT, ELIGIBILITY_CHECK, ELIGIBLE
+} = eligibilityURIs
+
+const IS_OWNER_OF_LAND = 'isOwnerOfLand'
+const HAS_LANDOWNER_PERMISSION = 'hasLandOwnerPermission'
+const PERMISSION_REQUIRED = 'permissionsRequired'
+const PERMISSION_GRANTED = 'permissionsGranted'
+
+// Helper to operate on the eligibility section of the journey cache
+export const eligibilityHelper = async (request, operator) => {
+  const journeyData = await request.cache().getData() || { eligibility: {} }
+  const { eligibility } = journeyData
+  operator(request, eligibility)
+  await request.cache().setData(journeyData)
+}
+
+// A state machine to determine the next page
+export const eligibilityCompletion = async request => {
+  const journeyData = await request.cache().getData() || { eligibility: {} }
+  const { eligibility } = journeyData
+  const grantedCompletionSection = e => {
```

Comment:
`e` is not a descriptive variable name, I'd suggest `eligibilityData` or something like that instead

Reply:
yes I didn't want extract those functions from the eligibilityCompletion function as they are completely contained but doing that caused sonar to moan about duplicate names in enclosing scopes. I have renamed to eligibilityPart.